src/clj/fluree/db/flake/reasoner.cljc

```diff
@@ -163,28 +165,47 @@
     (log/debug "Reasoner - source OWL rules: " graph)
     (owl-datalog/owl->datalog inserts graph)))
 
+(defn rules-from-dbs
+  [methods inserts dbs]
+  (go-try
+    (loop [[method & remaining-methods] methods
+           rules []]
+      (if method
+        (recur remaining-methods
+               (loop [[db & remaining-dbs] dbs
+                      rules* rules]
+                 (if db
+                   (recur remaining-dbs
+                          (into rules*
+                                (as-> db $
+                                  (<? (resolve/rules-from-db $ method))
+                                  (rules-from-graph method inserts $))))
+                   rules*)))
+        (remove empty? rules)))))
+
 (defn all-rules
   "Gets all relevant rules for the specified methods from the
   supplied rules graph or from the db if no graph is supplied."
-  [methods db inserts graph-or-db]
+  [methods db inserts rule-sources]
   (go-try
-    (let [rules-db        (cond
-                            (nil? graph-or-db) db
-                            (queryable? graph-or-db) graph-or-db)
-          supplied-rules* (when-not rules-db
-                            (try*
-                              (parse-rules-graph graph-or-db)
-                              (catch* e
-                                      (log/error "Error parsing supplied rules graph:" e)
-                                      (throw e))))]
-      (loop [[method & r] methods
-             rules []]
-        (if method
-          (let [rules-graph* (or supplied-rules*
-                                 (<? (resolve/rules-from-db rules-db method)))
-                rules*       (rules-from-graph method inserts rules-graph*)]
-            (recur r (into rules rules*)))
-          rules)))))
+    (let [rule-graphs           (filter #(and (map? %) (not (queryable? %))) rule-sources)
+          parsed-rule-graphs    (try*
+                                  (map parse-rules-graph rule-graphs)
+                                  (catch* e
+                                          (log/error "Error parsing supplied rules graph:" e)
+                                          (throw e)))
+          all-rules-from-graphs (mapcat (fn [method]
+                                          (mapcat (fn [parsed-rules-graph]
+                                                    (rules-from-graph method inserts parsed-rules-graph))
+                                                  parsed-rule-graphs))
+                                        methods)
+          rule-dbs              (filter #(or (string? %) (queryable? %)) rule-sources)
+          all-rule-dbs          (if (or (nil? rule-dbs) (empty? rule-dbs))
+                                  [db]
+                                  (conj rule-dbs db))
+          all-rules-from-dbs    (<? (rules-from-dbs methods inserts all-rule-dbs))
+          all-rules             (concat all-rules-from-graphs all-rules-from-dbs)]
```

Comment:
you should return the concatenation directly instead of making a new binding that you just return immediately.